aws-janitor/resources/s3_bucket.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"time"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type S3Bucket struct{}
+
+func (S3Bucket) MarkAndSweep(opts Options, set *Set) error {
+	logger := logrus.WithField("options", opts)
+	if !opts.EnableS3BucketClean {
+		logger.Info("Disable s3 bucket clean")
+		return nil
+	}
+	svc := s3.New(opts.Session, &aws.Config{Region: aws.String(opts.Region)})
+	resp, err := svc.ListBuckets(&s3.ListBucketsInput{})
+	if err != nil {
+		return err
+	}
+	for _, b := range resp.Buckets {
+		bucketName := *b.Name
+		if !strings.Contains(bucketName, opts.Region){
+			continue
+		}
+		respTags, err := svc.GetBucketTagging(&s3.GetBucketTaggingInput{Bucket: aws.String(bucketName)})
+		if err != nil {
+			continue
+		}
+		tags := fromS3Tags(respTags.TagSet)
+		if !set.Mark(opts, s3Bucket{bucketName: bucketName}, nil, tags) {
+			continue
+		}
+		if !opts.DryRun {
+			if _, err := svc.DeleteBucket(&s3.DeleteBucketInput{Bucket: &bucketName}); err != nil {
```

Comment:
It looks like only empty buckets can be deleted, do we need to delete objects first? https://docs.aws.amazon.com/AmazonS3/latest/API/API_DeleteBucket.html